File: src/ca/rightsomegoodgames/mayacharm/actions/SendBufferAction.java
package ca.rightsomegoodgames.mayacharm.actions;

import ca.rightsomegoodgames.mayacharm.mayacomms.MayaCommInterface;
import ca.rightsomegoodgames.mayacharm.settings.MCSettingsProvider;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.actionSystem.LangDataKeys;
import com.intellij.openapi.vfs.VirtualFile;
import org.jetbrains.annotations.NotNull;


public class SendBufferAction extends BaseSendAction {
    @Override
    public void actionPerformed(@NotNull AnActionEvent e) {
        final MCSettingsProvider settings = MCSettingsProvider.getInstance(e.getProject());
        final VirtualFile data = e.getData(LangDataKeys.VIRTUAL_FILE);
        if (data != null) {
            MayaCommInterface maya = new MayaCommInterface(settings.getHost(), settings.getPort());
            maya.sendFileToMaya(data.getPath());
        }
    }
}

File: src/ca/rightsomegoodgames/mayacharm/actions/SendBufferAction.kt
package ca.rightsomegoodgames.mayacharm.actions

import ca.rightsomegoodgames.mayacharm.MayaBundle as Loc
import ca.rightsomegoodgames.mayacharm.mayacomms.MayaCommandInterface
import ca.rightsomegoodgames.mayacharm.resources.MayaNotifications
import ca.rightsomegoodgames.mayacharm.settings.ProjectSettings
import com.intellij.notification.Notifications
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.actionSystem.LangDataKeys
import com.intellij.openapi.fileEditor.FileDocumentManager

class SendBufferAction : BaseSendAction(
        Loc.message("mayacharm.action.SendDocumentText"),
        Loc.message("mayacharm.action.SendDocumentDescription"), null) {
    override fun actionPerformed(e: AnActionEvent) {
        val sdk = ProjectSettings.getInstance(e.project!!).selectedSdk
        if (sdk == null) {
            Notifications.Bus.notify(MayaNotifications.NO_SDK_SELECTED)
            return
        }

        val docManager = FileDocumentManager.getInstance()
        val data = e.getData(LangDataKeys.VIRTUAL_FILE)

        val doc = data?.let { docManager.getDocument(it) }
        doc?.let {docManager.saveDocument(it)}

        if (data != null) {
            val maya = MayaCommandInterface(sdk.port)
            maya.sendFileToMaya(data.path)
        }
    }
}
